src/backend/PortabilityService.ConfigurationProvider/PortabilityServiceConfigurationExtensions.cs

```diff
@@ -10,13 +10,16 @@ namespace PortabilityService.ConfigurationProvider
 {
     public static class PortabilityServiceConfigurationExtensions
     {
-        public static IConfigurationBuilder AddPortabilityServiceConfiguration(this IConfigurationBuilder builder, string urlEnvironmentKeyName = Constants.UrlEnvironmentKeyName, string configurationSection = Constants.PortabilityServiceConfigurationRoot)
-            => AddPortabilityServiceConfiguration(builder, GetConfigurationServiceUrlFromEnvironment(urlEnvironmentKeyName), configurationSection);
+        public static IConfigurationBuilder AddPortabilityServiceConfiguration(this IConfigurationBuilder builder,
+                                                                               string UrlEnvironmentKeyName = ConfigurationProviderConstants.UrlEnvironmentKeyName,
+                                                                               string configurationSection = ConfigurationProviderConstants.PortabilityServiceConfigurationRoot,
+                                                                               bool optional = false)
+            => AddPortabilityServiceConfiguration(builder, GetConfigurationServiceUrlFromEnvironment(UrlEnvironmentKeyName), configurationSection, optional);
 
-        public static IConfigurationBuilder AddPortabilityServiceConfiguration(this IConfigurationBuilder builder, Uri configurationServiceUrl, string configurationSection = Constants.PortabilityServiceConfigurationRoot)
-            => AddPortabilityServiceConfiguration(builder, CreateHttpClient(configurationServiceUrl), configurationSection);
+        public static IConfigurationBuilder AddPortabilityServiceConfiguration(this IConfigurationBuilder builder, Uri configurationServiceUrl, string configurationSection = ConfigurationProviderConstants.PortabilityServiceConfigurationRoot, bool optional = false)
+            => AddPortabilityServiceConfiguration(builder, CreateHttpClient(configurationServiceUrl), configurationSection, optional);
 
-        public static IConfigurationBuilder AddPortabilityServiceConfiguration(this IConfigurationBuilder builder, HttpClient httpClient, string configurationSection = Constants.PortabilityServiceConfigurationRoot)
+        public static IConfigurationBuilder AddPortabilityServiceConfiguration(this IConfigurationBuilder builder, HttpClient httpClient, string configurationSection = ConfigurationProviderConstants.PortabilityServiceConfigurationRoot, bool optional = false)
```

Comment:
Instead of HttpClient, we may want to do HttpMessageHandler. Or, we can punt it for now, and use HttpClientFactory when that is released

Reply:
Let's wait for the ClientFactory